libraries/cyclestreets-core/src/main/java/net/cyclestreets/api/client/RetrofitApiClient.java

```diff
@@ -153,4 +159,29 @@ public UserJourneys getUserJourneys(final String username) throws IOException {
     Response<SendFeedbackResponseDto> response = v2Api.sendFeedback("routing", itinerary, comments, name, email).execute();
     return response.body().toFeedbackResult();
   }
+
+  public Upload.Result uploadPhoto(final String username,
+                                   final String password,
+                                   final double lon,
+                                   final double lat,
+                                   final long dateTime,
+                                   final String category,
+                                   final String metaCat,
+                                   final String caption,
+                                   final String filename) throws IOException {
+    MultipartBody.Part filePart = null;
+    if (filename != null) {
+      File file = new File(filename);
+      RequestBody fileBody = RequestBody.create(MediaType.parse("multipart/form-data"), file);
+      filePart = MultipartBody.Part.createFormData("mediaupload", file.getName(), fileBody);
+    }
+    // Unfortunately we have to do all this faff, otherwise the JSON converter will insert quotes!
+    RequestBody usernamePart = RequestBody.create(MediaType.parse("text/plain"), username);
```

Comment:
@mvl22 FYI, this was why we experienced some weirdness around email addresses - in my earlier efforts at migrating the user creation API (before [this commit](https://github.com/cyclestreets/android/commit/1ff3101c85de12e9318c99a2e67cb90736e39245#diff-1feaff77004e964c4335bda733e14addL33)) I was using multipart instead of form-urlencoded, and as a result the JSON converter was adding quotes around the strings.